### PR TITLE
Migrate to github.com/urfave/cli/v3

### PR DESCRIPTION
**Pull request by 6543**

### CODE REVIEW

**Comment by 6543** [comment]
Address https://github.com/woodpecker-ci/woodpecker/pull/3419 with tis too ...

**Comment by 6543** [comment]
-> always prefer _FILE

**Comment by anbraten** [comment]
Quite impatient looking forward for the feature persistent flags options / free ordering of flags in v3: https://github.com/urfave/cli/pull/1568

File: cmd/agent/main.go [hunk 1]
```diff
@@ -24,7 +26,7 @@ import (
 )
 
 func main() {
-	core.RunAgent([]backendTypes.Backend{
+	core.RunAgent(context.Background(), []backendTypes.Backend{
```
Comment (6543) [inline cmd/agent/main.go:29]: // TODO: test if we have to register signals for STRG-C ...

File: pipeline/backend/docker/docker.go [hunk 2]
```diff
@@ -63,7 +63,7 @@ func (e *docker) Name() string {
 }
 
 func (e *docker) IsAvailable(ctx context.Context) bool {
-	if c, ok := ctx.Value(backend.CliContext).(*cli.Context); ok {
+	if c, ok := ctx.Value(backend.CliContext).(*cli.Command); ok {
```
Comment (6543) [inline pipeline/backend/docker/docker.go:66]: test if injection still works!

**Comment by 6543** [comment]
ok @anbraten I'm waiting for https://github.com/urfave/cli/pull/1945 and if that's merged this here is as good as it can be

**Comment by 6543** [comment]
It's done ... just need to wait for the split out pulls to be merged first

File: pipeline/backend/types/config.go [hunk 3]
```diff
@@ -22,10 +22,10 @@ type Config struct {
 	Secrets  []*Secret  `json:"secrets"`  // secret definitions
 }
 
-// CliContext is the context key to pass cli context to backends if needed.
-var CliContext ContextKey
+// CliCommand is the context key to pass cli context to backends if needed.
+var CliCommand contextKey
 
-// ContextKey is just an empty struct. It exists so CliContext can be
+// ContextKey is just an empty struct. It exists so CliCommand can be
```
Comment (lafriks) [inline pipeline/backend/types/config.go:28]:
```suggestion
// contextKey is just an empty struct. It exists so CliCommand can be
```

**Review by lafriks — APPROVED** [review]
Small nit but otherwise lgtm

**Pull request merged**